web/federate.go

```diff
@@ -0,0 +1,98 @@
+package web
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"bitbucket.org/ww/goautoneg"
+	"github.com/golang/protobuf/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/text"
+
+	"github.com/prometheus/prometheus/promql"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+	dto "github.com/prometheus/client_model/go"
+)
+
+type Federation struct {
+	QueryEngine *promql.Engine
+}
+
+func (fed *Federation) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	req.ParseForm()
+	qs := fmt.Sprintf("{%s}", strings.Join(req.Form["match[]"], ","))
```

Comment:
What's `metric`? It just seems to be `__name__`.
Would be neat to have the label matching stuff be consistent across endpoints. I could see this working with `GET|DELETE /api/v1/series` as well, just that `match[]` now is a full metric selector and we OR between `match[]` values.